### PR TITLE
Option to remove sources in Physical Protection Job

### PR DESCRIPTION
**Pull request by naveena-maplelabs**
Changes:

1) New flag "delete_sources" is added to cohesity_job module.
2) If job state is "present" and flag is enabled, all the endpoints
   provided in the playbook is removed from job.
3) If flag is disabled, default behaviour occur(all endpoints are
   added to job)

### CODE REVIEW

File: library/cohesity_job.py [hunk 1]
```diff
@@ -582,7 +582,7 @@ def update_job(module, job_details, update_source_ids):
                    "user-agent": "Ansible-v2.2.0"}
         payload = job_details.copy()
         del payload['token']
-        if module.params.get('environment') == 'PhysicalFiles':
+        if module.params.get('environment') == 'PhysicalFiles' and module.params.get('delete_sources') == False:
```
Comment (pyashish) [inline library/cohesity_job.py:585]: Update the documentation for this. And if key doesn't exists default it to false ?

File: library/cohesity_job.py [hunk 2]
```diff
@@ -778,6 +778,34 @@ def update_vmware_job(module, job_meta_data, job_details):
         )
 
 
+def delete_sources(module, job_meta_data, job_details):
+    job_details['environment'] = 'Physical'
+    for source in module.params.get('protection_sources'):
+        job_details['endpoint'] = source['endpoint']
+        source_id = get__prot_source_id__by_endpoint(
```
Comment (pyashish) [inline library/cohesity_job.py]: Good stuff, we aren't telling users to supply ids, 

Can you put this in an try catch exception ?

File: library/cohesity_job.py [hunk 3]
```diff
@@ -778,6 +778,34 @@ def update_vmware_job(module, job_meta_data, job_details):
...
+        if source_id in job_meta_data['sourceIds']:
+            job_meta_data['sourceIds'].remove(source_id)
+        else:
+            continue
+        if module.params.get('environment') == 'PhysicalFiles':
+            for source in job_meta_data['sourceSpecialParameters']:
+                if source['sourceId'] == source_id:
+                    index = job_meta_data['sourceSpecialParameters'].index(source)
+                    del job_meta_data['sourceSpecialParameters'][index]
+    if len(job_meta_data['sourceIds']) == 0:
```
Comment (pyashish) [inline library/cohesity_job.py]: probably a good idea to add what failed.. say they are trying to remove 10 sources .. it partially failed. then it'd be useful. Else if its all or none update, then clearly explain to user why it failed.

File: tasks/job.yml [hunk 4]
```diff
@@ -19,4 +19,5 @@
     include: "{{ cohesity_protection.include | default('') }}"
     time_zone: "{{ cohesity_protection.time_zone | default('America/Los_Angeles') }}"
     start_time: "{{ cohesity_protection.start_time | default('') }}"
+    delete_sources:  "{{ cohesity_protection.delete_sources | default(False) }}"
```
Comment (pyashish) [inline tasks/job.yml:22]: On second thought, can't you create new role/task for this ? create , add sources/update, delete ,

File: library/cohesity_job.py (hunk 1 above)
Comment (naveena-maplelabs) [reply library/cohesity_job.py:585]: yeah

File: tasks/job.yml (hunk 4 above)
Comment (naveena-maplelabs) [reply tasks/job.yml:22]: Existing modules takes care of addition and updation, new changes is for deletion.

**Pull request merged**